src/main/java/com/google/sps/data/SmallCityService.java

```diff
@@ -28,10 +38,32 @@ public void findAllBusinesses() {
     businesses.add(new Listing("The Shiny Bean Coffee & Tea", new MapLocation(40.496328, -79.944862), 4.9, "https://goo.gl/maps/AYH2QCL7pkoMBxHA8", "https://theshinybean.com/"));
     businesses.add(new Listing("Weisshouse", new MapLocation(40.456684, -79.925499), 4.3, "https://goo.gl/maps/7tuXn7QF2hh7ioGYA", "https://www.weisshouse.com/"));
   }
+  
+  // Function is only here until the places API is implemented,
+  // and for test file to be able to set any static business LinkedList we want
+  // to try to use
+  public void setAllBusinesses(List<Listing> allBusinesses) {
+   businesses = allBusinesses;
+  }
 
-  public void eliminateBigBusinesses() {
-    // TODO: Parse big business list and remove big businesses from businessList
-
+  // Function to remove the big businesses from the list 
+  // that will be returned from the use of the Places API 
+  public void removeBigBusinessesFromResults() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("BigBusinesses");
+    PreparedQuery results = datastore.prepare(query);
+    Iterator <Listing> businessesList =  businesses.iterator();
+    Listing currentListing;
+    while (businessesList.hasNext()) {
+      currentListing = businessesList.next();
+      for (Entity entity : results.asIterable()) {
+        String businessName = (String) entity.getProperty("business");
+        if(businessName.equals(currentListing.getName())) {
+          businessesList.remove();
+          break;
```

Comment:
Is the "break" necessary"?

Reply:
I just want it so that second the business name is the same as the big business in the database, no need to continue checking the names in the database.

Reply:
I have tried to write a while loop, but currently it is not working to what I expected it to do. I want to stick with this solution for the time being, what do you think?